### PR TITLE
Add pushes to CI, limited to tags starting with 'v', for releasing

### PR DESCRIPTION
**Pull request by pinin4fjords**
Tiny fix- we need to run the workflow on pushes as well as pull requests, for release tags only.

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: Python package
 
-on: [pull_request]
+on: 
+    pull_request:
+    push:
+        tags:
+            - v*
```
Comment (pcm32) [inline .github/workflows/python-package.yml:7]: maybe easier just to point to master for pushes?

**Review by pcm32 — APPROVED** [review]
Looks good to me. A small suggestion, but not required. Only tagged releases make it to master under the the branching model.

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (pinin4fjords) [reply .github/workflows/python-package.yml]: Well, I don't want the release steps to trigger for any old push against master- just for tags.

File: .github/workflows/python-package-release.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+on: 
+    push:
+        tags:
+            - v*
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (pcm32) [inline .github/workflows/python-package-release.yml:11]: I could imagine that you could find containers here that have pip and python already installed, which could speed things up.

**Review by pcm32 — APPROVED** [review]
LGTM, some suggestion on the container to use.

File: .github/workflows/python-package-release.yml (hunk 2 above)
Comment (pinin4fjords) [reply .github/workflows/python-package-release.yml:11]: setup-python seems to be what pretty much everyone does: https://github.com/actions/setup-python. I think it's doing some caching anyway, seems pretty quick to start.

**Comment by pinin4fjords** [comment]
Merging for now, expecting improvements later

**Comment by pinin4fjords** [comment]
Note: release process seems to work as well: https://pypi.org/project/scanpy-scripts/

**Pull request merged**